tests/tools/clean_orphans_gcp.py

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env python3
+
+import argparse
+import re
+import os
+import sys
+
+import google.cloud.compute as compute
+import google.cloud.storage as storage
+import google.oauth2
+
+
+parser = argparse.ArgumentParser(description = 'Cleanup integration test resources in GCP')
+parser.add_argument('-p', '--project', metavar='project_id', required=False, dest='project', default='sap-cp-k8s-gdnlinux-gcp-test',
+                    help='Default value is "sap-cp-k8s-gdnlinux-gcp-test"')
+parser.add_argument('-r', '--region', metavar='region', required=False, dest='region', default='europe-west1', 
+                    help='Default value is "europe-west1"')
+parser.add_argument('-z', '--zone', metavar='zone', required=False, dest='zone', default='europe-west1-d',
+                    help='Default value is "europe-west1-d"')
+parser.add_argument('-a', '--auth', metavar='service_account.json', required=False, dest='auth_path', default='',
+                    help='Path to JSON file that holds the service account login credentials')
+args = parser.parse_args()
+
+if os.path.exists(args.auth_path):
+    credentials = google.oauth2.service_account.Credentials.from_service_account_file(args.auth_path)
+else:
+    credentials = None
```

Comment:
When falling back to default / well-known paths, we should validate for missing credentials and/or catch `google.auth.exceptions.DefaultCredentialsError`.